Logs to hdd

swbsf opened the pull request:
Here is an example on how to solve issues we got writing to SD card. But please beware : 

1. - This will apply to any Koombook on field (we could for example enable this on a project basis).
2. - If applied as is, old logs will be "lost", as in inaccessible (changes applied after next reboot).
3. - We may have a broader look at which folder may be missing (modoboa?). Most apps aren't recreating their log folders in /var/log if it is missing.
4. - As it is, it's roughly tested. The idea is to give a simple base to talk about. My first opinion after writing this is : it should be included in Koombook system image, not run by ansible (see third point).

Commit:
Necessary work to write logs to hdd

Commit:
Necessary work to write logs to hdd

Commit:
Doing this instead of disabling mount task in role/system as mount point in created same time fstab is edited.
Otherwise we would need to reboot Koombook right away after installing these units.

Commit:
Doing this instead of disabling mount task in role/system as mount point in created same time fstab is edited.
Otherwise we would need to reboot Koombook right away after installing these units.

Commit:
mount units

Commit:
mount units

Commit:
some linting

Commit:
some linting

Commit:
Including task.

Commit:
Including task.

Commit:
Adding symlinks to basic.target.wants.
This will ensure mounts will be created before multi-user.target units are started.

Commit:
Adding symlinks to basic.target.wants.
This will ensure mounts will be created before multi-user.target units are started.

Commit:
Add kb-usa-brainerd project

Commit:
Add kb-cog-rfi project

Commit:
Fix project name

Commit:
Add packages for kb-usa-brainerd devices

Commit:
fixing typo.

Commit:
fixing typo.

Commit:
Enable use of external antenna

This new tasks use a script to detect if the device got an IP address
given by a dhcp server,
if true, the device is within a local network probably connected to
Internet. If the device does not get any IP addresss but an ethernet
link is detected we asign an arbitrary IP address to the interface and
we assume that an external Ubiquiti wifi antenna is connected and is
waiting for an IP address from given by our device.

Commit:
cosmetic

Commit:
specify on which interface to listen to, except is not usefull here

Commit:
Associate hostname and IP of the new interface for external wifi antenna

Commit:
Declare use of external-antenna in device_list

Commit:
Declare var and IP for new network interface

Commit:
dnsmasq answere on 192.168.3.1 as well

Commit:
Use a specific network for the external antenna

Commit:
dsnmasq must listen on wlan0 or wlan0,eth0 depending on the situation

Commit:
reload is not sufficient, restart needed

Commit:
Fix json

Commit:
Typo, use underscore

Commit:
Typo, use underscore

Commit:
Enable external antenna script only when asked for

Commit:
Add default value for ideascube_package_name

Commit:
Give some flexibility to ideascube playbook

I replaced ideascube package name with the variable
ideascube_package_name defined in device_list to have the ability to install other ideascube
type of package, RFI for instance.

This finaly does not modify too many tasks and most of them stays
exactly the same.

Commit:
Better readability on any physical device.

Commit:
Updating package list for idb-fra-rambouillet

Commit:
Add dhcp-range for network 192.168.3.0/24

fheslouin commented on roles/logs/tasks/logs_to_hdd.yml:
```diff
@@ -0,0 +1,57 @@
+      owner: root
+      group: adm
+    - name: telegraf
+      owner: telegraf
```
At this stage the user `telegraf` is unknown. Telegraf role should be executed before logs role

Commit:
Add the .mount extension

Commit:
Add the .mount extension

Commit:
Add missing [Install] section needed for unit activation

Commit:
Add missing [Install] section needed for unit activation

Commit:
Add missing log folder

Commit:
Add missing log folder

Commit:
Swap path for dest

Commit:
Swap path for dest

Commit:
Move up telegraf role, otherwhise user is missing for the next role

Commit:
Move up telegraf role, otherwhise user is missing for the next role

Commit:
Create app folder in uwsgi as well

Commit:
Create app folder in uwsgi as well

Commit:
Change execution level for logs role

"Logs" role must be executed mostly at master and update stage, however the
log_to_hdd.yml tasks must be executed only at master stage. We don't
want devices on field to get this feature as it implies quite tricky changes.

Commit:
Change execution level for logs role

"Logs" role must be executed mostly at master and update stage, however the
log_to_hdd.yml tasks must be executed only at master stage. We don't
want devices on field to get this feature as it implies quite tricky changes.

Commit:
Change execution level for telegraf, must be installed before logs role is executed

Commit:
Change execution level for telegraf, must be installed before logs role is executed

fheslouin committed:
Merge pull request #176 from ideascube/captiveportal_readability

Better readability on any physical device.

Commit:
Add external_hotspot_ip variable to iptables rules

letompouce committed:
Conf draft for idb-ita-palermo

Commit:
update doc & KoomBook linux image link

Commit:
typo

Commit:
Adding missing package films-pour-enfants-fr

fheslouin committed:
Merge pull request #173 from ideascube/use-external-antenna

[WIP] Use external antenna

fheslouin committed:
Merge pull request #174 from ideascube/ideascube-rfi

Adapt ideascube to irfi-ideascube package

Commit:
Merge branch 'logs_to_hdd' of github.com:ideascube/ansiblecube into logs_to_hdd

Commit:
Split logs_to_hdd tasks

I had to move some ansible tasks to a bash script executed at the first
boot. The main reason is that the HDD might not be connected when the
master tag is executed. Thus, HDD is mounted when the device start for
the first start, script koombook_zeroconf take care of this, it then
synchronise the log folder and so on.

Commit:
Uncomment wget line

fheslouin commented:
This PR become a bit of a mess, re-opened and cleaned up on #178
